docs/src/main/asciidoc/cache-infinispan-reference.adoc

```diff
@@ -83,10 +83,82 @@ quarkus.cache.infinispan.client-name=another
 
 == Marshalling
 
-When interacting with Infinispan in Quarkus, you can easily marshal and unmarshal
-Java simple types when writing to or reading from the cache. However, when dealing
-with Plain Old Java Objects (POJOs), users of Infinispan need to provide the marshalling
-schema.
+When interacting with Infinispan in Quarkus, you can easily serialize and deserialize Java primitive types when storing or retrieving data from the cache. No additional marshalling configuration is required for Infinispan.
+
+[source, java]
+----
+@CacheResult(cacheName = "weather-cache") //<1>
+public String getDailyForecast(String dayOfWeek, int dayOfMonth, String city) { //<2>
+    return dayOfWeek + " will be " + getDailyResult(dayOfMonth % 4) + " in " + city;
+}
+----
+<1> Ask this method execution to be cached in the 'weather-cache'
+<2> The key combines `String` dayOfWeek, `int` dayOfMonth and `String` city. The associated value is of type `String`.
+
+However, when using Plain Old Java Objects (POJOs), users of Infinispan must provide the marshalling schema.
```

Comment:
Should you mention that it is Protobuf?

Reply:
I do mention it but later on, I'm going to mention it earlier. thanks!

Reply:
@cescoffier I updated this part with the explanation